alcs-frontend/src/app/shared/header/header.component.ts

```diff
@@ -12,27 +16,109 @@ import { AuthenticationService } from '../../services/authentication/authenticat
 })
 export class HeaderComponent implements OnInit {
   homeUrl = environment.homeUrl;
-  decisionMakers: ApplicationDecisionMakerDto[] = [];
+  currentUser!: ICurrentUser;
+  currentUserProfile?: UserDto;
+  private decisionMakers: ApplicationDecisionMakerDto[] = [];
+  $sortedDecisionMakers!: Observable<ApplicationDecisionMakerDto[]>;
 
   constructor(
     private authService: AuthenticationService,
     private applicationService: ApplicationService,
+    private toastService: ToastService,
+    private userService: UserService,
     private router: Router
   ) {}
 
   ngOnInit(): void {
-    this.applicationService.$applicationDecisionMakers.subscribe((dms) => {
-      this.decisionMakers = dms;
+    this.authService.isAuthenticated.subscribe((isAuthenticated) => {
+      if (isAuthenticated) {
+        this.currentUser = this.authService.getCurrentUser();
+
+        this.userService.fetchUsers();
+        this.applicationService.setup();
+      }
+    });
+
+    this.userService.$users.subscribe((users) => {
+      this.currentUserProfile = users.find((u) => u.email === this.currentUser.email);
     });
+
+    this.$sortedDecisionMakers = this.applicationService.$applicationDecisionMakers.pipe(
+      map((dms) => this.transformDecisionMakers(dms))
+    );
+  }
+
+  private transformDecisionMakers(dms: ApplicationDecisionMakerDto[]) {
```

Comment:
Since favourites also needs to be indicated on the board component, should this enrichment be done in the application service?

The application service could have the subscribe and this logic, this would mean that isFavourite is accessible in all components instead of just the header.

Reply:
Sure, I'll add it on Board page, when I started this ticket I did not have titles on board at all 😅